src/main/java/impl/ui/Parser.java

```diff
@@ -0,0 +1,221 @@
+            case Unmark:
+                handleUnmark(in);
+                break;
+            case Delete:
+                handleDelete(in);
+                break;
+            case Find:
+                handleFind(in);
+                break;
+            case Error:
+                System.out.println("Error! Please try again :(");
+                break;
+            case Unknown:
+                // Fall through as default
+            default:
+                System.out.println("Unknown Command Detected!");
+                break;
+        }
+    }
+    private void handleList(){
+        System.out.println("These are all your unfinished tasks:");
+        String text;
+        for (int i = 0; i < list.size(); i++) {
+            text = (i+1) + "." + list.get(i).toString();
+            System.out.println(text);
+        }
+    }
+
+    private void handleToDo(String in){
+        String description = in.substring(5);
+        Task add = new ToDos(description);
+        list.add(add);
+        System.out.println("added: " + add.toString());
+    }
+    private void handleDeadline(String in){
+        String[] deadlineSplit = in.split(" /by ");
+        String description = deadlineSplit[0].substring(9);
+        String by = handleDate(deadlineSplit[1]);
+        Task add = new Deadlines(description,by);
+        list.add(add);
+        System.out.println("added: " + add.toString());
+    }
+
+    private void handleEvent(String in){
+        String[] eventSplit = in.split(" /from ");
+        String description = eventSplit[0].substring(6);
+        String from = handleDate(eventSplit[1].split(" /to ")[0]);
+        String by = handleDate(eventSplit[1].split("/to ")[1]);
+        Task add = new Events(description,by,from);
+        list.add(add);
+        System.out.println("added: " + add.toString());
+    }
+
+    private void handleMark(String in){
+        int i = Integer.parseInt(in.split(" ")[1])-1;
+        System.out.println("I have marked this task as done");
+        list.get(i).setDone();
+        String text = (i+1) + "." + list.get(i).toString();
+        System.out.println(text);
+    }
+
+    private void handleUnmark(String in){
+        int i = Integer.parseInt(in.split(" ")[1])-1;
+        list.get(i).setUnDone();
+        System.out.println("I have marked this task as undone");
+        String text = (i+1) + "." + list.get(i).toString();
+        System.out.println(text);
+    }
+
+    private void handleDelete(String in){
+        int i = Integer.parseInt(in.split(" ")[1])-1;
+        System.out.println("I have removed this task from the list");
+        String text = (i+1) + "." + list.get(i).toString();
+        System.out.println(text);
+        list.remove(i);
+    }
+    private void handleFind(String in){
+        String find = in.split(" ",2)[1];
+        String text;
+        for (int i = 0; i < list.size(); i++) {
+            if(list.get(i).getDescription().toLowerCase().contains(find.toLowerCase())){
```

Comment:
I think this line might be too long, you may consider extending it to the next line.